packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -281,22 +275,56 @@ export const Product = ({
     // if desired product is already subscribed to, show iap already subscribed error
     // else, product is not subscribed to, but on same product set/might be eligible for upgrade
     // show iap upgrade contact support error messaging
-    const iapErrorMessageCode = () => {
-      return iapSubscription
-        ? 'iap_already_subscribed' // already subscribed to this product
-        : 'iap_upgrade_contact_support'; // different product, but same product set/eligible for upgrade
-    };
-
     if (planUpgradeEligibility === 'blocked_iap') {
+      // Get plan customer is blocked on
+      const currentPlan = () => {
```

Comment:
Instead of performing this logic on the frontend, would it better to update the `plan-eligibility` endpoint?

@julianpoy what do you think?

Reply:
Just some additional context, and @sardesam correct me if I'm wrong here. The front-end needs to know which plan the customer is currently subscribed to, that's causing the blockage. That's what this function is supposed to determine.

However, instead of doing it here, could we update the `capabilities/plan-eligibility` API to return that info instead?

Reply:
Agreed. I tried to find something that returned the affected plan specifically (and for IAP). I wrote this up for the dot release but a fast follow to get this logic, or similar (this function only checks IAP subscriptions), moved over to the backend and available everywhere would be really nice.

Reply:
After discussing with @chelsea-lewis and @StaberindeZA , updated FXA-6166 to include moving this to the backend. 👍